final_project.py

```
@@ -0,0 +1,196 @@
+
+col = df2.columns
+col
+
+X_sca = df2[col]
+
+import statsmodels.api as sm
+X_sca = sm.add_constant(X_sca)
+
+from statsmodels.stats.outliers_influence import variance_inflation_factor
+
+X_numeric = X.select_dtypes(include=['int64', 'float64'])  # Keep only numeric columns
```

Ruff (0.12.2)

Undefined name `X`

(F821)